### ISSUE TITLE
[Bug]: InsertLeave is broken when copilot used as chat adapter

### ISSUE THREAD

**Issue by dhruvinsh** [olimorris/codecompanion.nvim#176]
### Your `minimal.lua` config

```lua
    strategies = {
      chat = { adapter = "copilot" },
      inline = { adapter = "copilot" },
      agent = { adapter = "copilot" },
    },
```

### Error messages

```
Error detected while processing InsertLeave Autocommands for "<buffer=8>":
Error executing lua callback: vim/shared.lua:0: t: expected table, got nil
stack traceback:
        [C]: in function 'error'
        vim/shared.lua: in function 'validate'
        vim/shared.lua: in function 'tbl_contains'
        ...vim/lazy/codecompanion.nvim/lua/codecompanion/schema.lua:46: in function 'validate_type'
        ...vim/lazy/codecompanion.nvim/lua/codecompanion/schema.lua:82: in function 'validate_field'
        ...vim/lazy/codecompanion.nvim/lua/codecompanion/schema.lua:99: in function 'validate'
        ...codecompanion.nvim/lua/codecompanion/strategies/chat.lua:512: in function <...codecompanion.nvim/lua/codecompanion/strategies/chat.lua:509>
```

### Describe the bug

For `copilot` adapter schema, there are no `choices` define. [lua/codecompanion/adapters/copilot.lua](https://github.com/olimorris/codecompanion.nvim/blob/b7db7ea0b58dcab69aa7e4d709283ba8e718a8d8/lua/codecompanion/adapters/copilot.lua#L121-L158)

And due to that [valida_type](https://github.com/olimorris/codecompanion.nvim/blob/b7db7ea0b58dcab69aa7e4d709283ba8e718a8d8/lua/codecompanion/schema.lua#L37-L46) funtion can't do `vim.tbl_contains(choices, value)` check since `choices` will be `nil` value in case of copilot.

All of these breaks [InsertLeave](https://github.com/olimorris/codecompanion.nvim/blob/b7db7ea0b58dcab69aa7e4d709283ba8e718a8d8/lua/codecompanion/strategies/chat.lua#L505-L536) auto command.

### Final checks

- [X] I have made sure this issue exists in the latest version of the plugin
- [X] I have tested with the `minimal.lua` file from above and have shared this
- [X] I have shared the contents of the log file

**Comment by olimorris** [comment]
Assume you have show settings enabled as well?

**Comment by olimorris** [comment]
Thanks for raising this. It's now fixed in 2.2.5
